Assets/Editor/GroundhogEditorMenu.cs

```diff
@@ -10,4 +11,70 @@ static void MovePlayerHere()
     {
         GameObject.FindWithTag("Player").transform.position = SceneView.lastActiveSceneView.camera.transform.position;
     }
+
+     [MenuItem("Assets/Create Reversed Clip", false, 14)]
+    private static void ReverseClip()
+    {
+        string directoryPath = Path.GetDirectoryName(AssetDatabase.GetAssetPath(Selection.activeObject));
+        string fileName = Path.GetFileName(AssetDatabase.GetAssetPath(Selection.activeObject));
+        string fileExtension = Path.GetExtension(AssetDatabase.GetAssetPath(Selection.activeObject));
+        fileName = fileName.Split('.')[0];
+        string copiedFilePath = directoryPath + Path.DirectorySeparatorChar + fileName + "_Reversed" + fileExtension;
+        var clip = GetSelectedClip();
+ 
+        AssetDatabase.CopyAsset(AssetDatabase.GetAssetPath(Selection.activeObject), copiedFilePath);
+ 
+        clip  = (AnimationClip)AssetDatabase.LoadAssetAtPath(copiedFilePath, typeof(AnimationClip));
+ 
+        if (clip == null)
+            return;
+        float clipLength = clip.length;
+        var curves = AnimationUtility.GetAllCurves(clip, true);
```

Comment:
this line causes a warning, although our product has a few of those for sure

Reply:
Yeah, Unity doesn't like deprecated methods being used, even though they still work.